spyder/widgets/variableexplorer/collectionseditor.py

```diff
@@ -511,6 +514,17 @@ def create_dialog(self, editor, data):
                      lambda eid=id(editor): self.editor_rejected(eid))
         editor.show()
         
+    @Slot(str, object)
+    def change_option(self, option_name, new_value):
+        """
+        Change configuration option.
+
+        This function is called when a `sig_option_changed` signal is received.
+        At the moment, this signal can only come from a DataFrameEditor.
+        """
+        assert option_name == 'dataframe_format'
+        self.parent().set_dataframe_format(new_value)

```

Comment:
The `collectionseditor` creates the `dataframeeditor` so it has to inject the relevant configuration options (here only the dataframe format) in the `dataframeeditor`.

As I understand the widget/plugin separation, widgets should get all relevant configuration options at creation time. Widgets are not supposed to query the corresponding plugin because it should be possible to run them stand alone. In this instance, the `collectioneditor` needs to keep track of the dataframe format, so I don't see how to move this method to `dataframeeditor`.